pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -556,3 +556,53 @@ func (r *reconcilerBase) isLastReconciled(nn types.NamespacedName, resourceVersi
 	}
 	return resourceVersion == lastReconciled
 }
+
+func (r *reconcilerBase) upsertNotificationCR(ctx context.Context, rs client.Object) (*v1beta1.Notification, error) {
```

Comment:
Suggest adding labels to the notification CR for querying with labelSelector
```
configsync.gke.io/sync-namespace: ...
configsync.gke.io/sync-name: ...
configsync.gke.io/sync-kind: ...
```

Reply:
I'm not opposed, but I'm curious about the value add. Given the Notification has the same name/namespace as the RSync (and the namespace implies the sync-kind), it seems like a user could query using name/namespace. I suppose perhaps it's an implementation detail that the name/namespace is the same, and that might not be transparent to the user.

Did you have a use case in mind?

Reply:
The use case I can think of is to filter certain Notification CRs with the labels via CLI/UI.

After a second thought, since the CR has the same name/namespace as the RSync, we can also use the namespace/name to select the objects. For UI, it is most likely the Notification status is just a column of the Package, rather than a separate page.

I think it is okay to not add the labels for now.